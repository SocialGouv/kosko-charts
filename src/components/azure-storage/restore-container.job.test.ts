import { EnvVar } from "kubernetes-models/v1/EnvVar";

import { restoreContainerJob } from "./restore-container.job";

process.env.CI_COMMIT_SHORT_SHA = "b123a99";

test("should create restore container job", () => {
  expect(
    restoreContainerJob({
      env: [
        new EnvVar({
          name: "SOURCE_CONTAINER",
          value: "prod-files",
        }),
        new EnvVar({
          name: "DESTINATION_CONTAINER",
          value: "preprod-files",
        }),
      ],
      project: "some-cool-project",
    })
  ).toMatchSnapshot();
});
