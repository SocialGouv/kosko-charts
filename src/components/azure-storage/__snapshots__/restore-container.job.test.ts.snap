// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should create restore job from dev to dev 1`] = `
Object {
  "apiVersion": "batch/v1",
  "kind": "Job",
  "metadata": Object {
    "annotations": Object {
      "kapp.k14s.io/update-strategy": "skip",
    },
    "labels": Object {
      "application": "www",
      "component": "restore-container",
      "owner": "my-team",
      "team": "my-team",
    },
    "name": "restore-container-my-test-branch-47n9d8",
    "namespace": "some-cool-project-secret",
  },
  "spec": Object {
    "backoffLimit": 0,
    "template": Object {
      "metadata": Object {
        "annotations": Object {
          "kapp.k14s.io/deploy-logs": "for-new-or-existing",
        },
      },
      "spec": Object {
        "containers": Array [
          Object {
            "command": Array [
              "sh",
              "-c",
              "

[ ! -z $SOURCE_CONTAINER ] || (echo \\"No SOURCE_CONTAINER\\"; exit 1)
[ ! -z $SOURCE_ACCOUNT_NAME ] || (echo \\"No SOURCE_ACCOUNT_NAME\\"; exit 1)
[ ! -z $SOURCE_ACCOUNT_KEY ] || (echo \\"No SOURCE_ACCOUNT_KEY\\"; exit 1)
[ ! -z $DESTINATION_CONTAINER ] || (echo \\"No DESTINATION_CONTAINER\\"; exit 1)
[ ! -z $DESTINATION_ACCOUNT_NAME ] || (echo \\"No DESTINATION_ACCOUNT_NAME\\"; exit 1)
[ ! -z $DESTINATION_ACCOUNT_KEY ] || (echo \\"No DESTINATION_ACCOUNT_KEY\\"; exit 1)


echo \\"starting restore container $SOURCE_CONTAINER into $DESTINATION_CONTAINER\\"

# copy container
az storage copy -r   --source-account-name \\"$SOURCE_ACCOUNT_NAME\\"   --source-account-key \\"$SOURCE_ACCOUNT_KEY\\"   --source-container \\"$SOURCE_CONTAINER\\"   --account-name \\"$DESTINATION_ACCOUNT_NAME\\"   --account-key \\"$DESTINATION_ACCOUNT_KEY\\"   --destination-container \\"$DESTINATION_CONTAINER\\"

# get original permission
PUBLIC_ACCESS=\`az storage container show-permission   --account-name \\"$SOURCE_ACCOUNT_NAME\\"   --account-key \\"$SOURCE_ACCOUNT_KEY\\"   --name $SOURCE_CONTAINER | jq -r \\".publicAccess\\"\`

# set original permission
az storage container set-permission   --name $DESTINATION_CONTAINER   --account-name \\"$DESTINATION_ACCOUNT_NAME\\"   --account-key \\"$DESTINATION_ACCOUNT_KEY\\"   --public-access $PUBLIC_ACCESS

",
            ],
            "env": Array [
              Object {
                "name": "SOURCE_ACCOUNT_NAME",
                "valueFrom": Object {
                  "secretKeyRef": Object {
                    "key": "azurestorageaccountname",
                    "name": "azure-somecoolprojectdev-volume",
                  },
                },
              },
              Object {
                "name": "SOURCE_ACCOUNT_KEY",
                "valueFrom": Object {
                  "secretKeyRef": Object {
                    "key": "azurestorageaccountkey",
                    "name": "azure-somecoolprojectdev-volume",
                  },
                },
              },
              Object {
                "name": "DESTINATION_ACCOUNT_NAME",
                "valueFrom": Object {
                  "secretKeyRef": Object {
                    "key": "azurestorageaccountname",
                    "name": "azure-somecoolprojectdev-volume",
                  },
                },
              },
              Object {
                "name": "DESTINATION_ACCOUNT_KEY",
                "valueFrom": Object {
                  "secretKeyRef": Object {
                    "key": "azurestorageaccountkey",
                    "name": "azure-somecoolprojectdev-volume",
                  },
                },
              },
              Object {
                "name": "SOURCE_CONTAINER",
                "value": "dev-files1",
              },
              Object {
                "name": "DESTINATION_CONTAINER",
                "value": "dev-files2",
              },
            ],
            "envFrom": Array [],
            "image": "mcr.microsoft.com/azure-cli:2.15.1",
            "imagePullPolicy": "IfNotPresent",
            "name": "restore-db",
            "resources": Object {
              "limits": Object {
                "cpu": "300m",
                "memory": "512Mi",
              },
              "requests": Object {
                "cpu": "50m",
                "memory": "64Mi",
              },
            },
          },
        ],
        "restartPolicy": "Never",
      },
    },
    "ttlSecondsAfterFinished": 86400,
  },
}
`;

exports[`should create restore job from dev to prod 1`] = `
Object {
  "apiVersion": "batch/v1",
  "kind": "Job",
  "metadata": Object {
    "annotations": Object {
      "kapp.k14s.io/update-strategy": "skip",
    },
    "labels": Object {
      "application": "www",
      "component": "restore-container",
      "owner": "my-team",
      "team": "my-team",
    },
    "name": "restore-container-my-test-branch-47n9d8",
    "namespace": "some-cool-project-secret",
  },
  "spec": Object {
    "backoffLimit": 0,
    "template": Object {
      "metadata": Object {
        "annotations": Object {
          "kapp.k14s.io/deploy-logs": "for-new-or-existing",
        },
      },
      "spec": Object {
        "containers": Array [
          Object {
            "command": Array [
              "sh",
              "-c",
              "

[ ! -z $SOURCE_CONTAINER ] || (echo \\"No SOURCE_CONTAINER\\"; exit 1)
[ ! -z $SOURCE_ACCOUNT_NAME ] || (echo \\"No SOURCE_ACCOUNT_NAME\\"; exit 1)
[ ! -z $SOURCE_ACCOUNT_KEY ] || (echo \\"No SOURCE_ACCOUNT_KEY\\"; exit 1)
[ ! -z $DESTINATION_CONTAINER ] || (echo \\"No DESTINATION_CONTAINER\\"; exit 1)
[ ! -z $DESTINATION_ACCOUNT_NAME ] || (echo \\"No DESTINATION_ACCOUNT_NAME\\"; exit 1)
[ ! -z $DESTINATION_ACCOUNT_KEY ] || (echo \\"No DESTINATION_ACCOUNT_KEY\\"; exit 1)


echo \\"starting restore container $SOURCE_CONTAINER into $DESTINATION_CONTAINER\\"

# copy container
az storage copy -r   --source-account-name \\"$SOURCE_ACCOUNT_NAME\\"   --source-account-key \\"$SOURCE_ACCOUNT_KEY\\"   --source-container \\"$SOURCE_CONTAINER\\"   --account-name \\"$DESTINATION_ACCOUNT_NAME\\"   --account-key \\"$DESTINATION_ACCOUNT_KEY\\"   --destination-container \\"$DESTINATION_CONTAINER\\"

# get original permission
PUBLIC_ACCESS=\`az storage container show-permission   --account-name \\"$SOURCE_ACCOUNT_NAME\\"   --account-key \\"$SOURCE_ACCOUNT_KEY\\"   --name $SOURCE_CONTAINER | jq -r \\".publicAccess\\"\`

# set original permission
az storage container set-permission   --name $DESTINATION_CONTAINER   --account-name \\"$DESTINATION_ACCOUNT_NAME\\"   --account-key \\"$DESTINATION_ACCOUNT_KEY\\"   --public-access $PUBLIC_ACCESS

",
            ],
            "env": Array [
              Object {
                "name": "SOURCE_ACCOUNT_NAME",
                "valueFrom": Object {
                  "secretKeyRef": Object {
                    "key": "azurestorageaccountname",
                    "name": "azure-somecoolprojectdev-volume",
                  },
                },
              },
              Object {
                "name": "SOURCE_ACCOUNT_KEY",
                "valueFrom": Object {
                  "secretKeyRef": Object {
                    "key": "azurestorageaccountkey",
                    "name": "azure-somecoolprojectdev-volume",
                  },
                },
              },
              Object {
                "name": "DESTINATION_ACCOUNT_NAME",
                "valueFrom": Object {
                  "secretKeyRef": Object {
                    "key": "azurestorageaccountname",
                    "name": "azure-somecoolprojectprod-volume",
                  },
                },
              },
              Object {
                "name": "DESTINATION_ACCOUNT_KEY",
                "valueFrom": Object {
                  "secretKeyRef": Object {
                    "key": "azurestorageaccountkey",
                    "name": "azure-somecoolprojectprod-volume",
                  },
                },
              },
              Object {
                "name": "SOURCE_CONTAINER",
                "value": "dev-files1",
              },
              Object {
                "name": "DESTINATION_CONTAINER",
                "value": "prod-files2",
              },
            ],
            "envFrom": Array [],
            "image": "mcr.microsoft.com/azure-cli:2.15.1",
            "imagePullPolicy": "IfNotPresent",
            "name": "restore-db",
            "resources": Object {
              "limits": Object {
                "cpu": "300m",
                "memory": "512Mi",
              },
              "requests": Object {
                "cpu": "50m",
                "memory": "64Mi",
              },
            },
          },
        ],
        "restartPolicy": "Never",
      },
    },
    "ttlSecondsAfterFinished": 86400,
  },
}
`;

exports[`should create restore job from prod to dev 1`] = `
Object {
  "apiVersion": "batch/v1",
  "kind": "Job",
  "metadata": Object {
    "annotations": Object {
      "kapp.k14s.io/update-strategy": "skip",
    },
    "labels": Object {
      "application": "www",
      "component": "restore-container",
      "owner": "my-team",
      "team": "my-team",
    },
    "name": "restore-container-my-test-branch-47n9d8",
    "namespace": "some-cool-project-secret",
  },
  "spec": Object {
    "backoffLimit": 0,
    "template": Object {
      "metadata": Object {
        "annotations": Object {
          "kapp.k14s.io/deploy-logs": "for-new-or-existing",
        },
      },
      "spec": Object {
        "containers": Array [
          Object {
            "command": Array [
              "sh",
              "-c",
              "

[ ! -z $SOURCE_CONTAINER ] || (echo \\"No SOURCE_CONTAINER\\"; exit 1)
[ ! -z $SOURCE_ACCOUNT_NAME ] || (echo \\"No SOURCE_ACCOUNT_NAME\\"; exit 1)
[ ! -z $SOURCE_ACCOUNT_KEY ] || (echo \\"No SOURCE_ACCOUNT_KEY\\"; exit 1)
[ ! -z $DESTINATION_CONTAINER ] || (echo \\"No DESTINATION_CONTAINER\\"; exit 1)
[ ! -z $DESTINATION_ACCOUNT_NAME ] || (echo \\"No DESTINATION_ACCOUNT_NAME\\"; exit 1)
[ ! -z $DESTINATION_ACCOUNT_KEY ] || (echo \\"No DESTINATION_ACCOUNT_KEY\\"; exit 1)


echo \\"starting restore container $SOURCE_CONTAINER into $DESTINATION_CONTAINER\\"

# copy container
az storage copy -r   --source-account-name \\"$SOURCE_ACCOUNT_NAME\\"   --source-account-key \\"$SOURCE_ACCOUNT_KEY\\"   --source-container \\"$SOURCE_CONTAINER\\"   --account-name \\"$DESTINATION_ACCOUNT_NAME\\"   --account-key \\"$DESTINATION_ACCOUNT_KEY\\"   --destination-container \\"$DESTINATION_CONTAINER\\"

# get original permission
PUBLIC_ACCESS=\`az storage container show-permission   --account-name \\"$SOURCE_ACCOUNT_NAME\\"   --account-key \\"$SOURCE_ACCOUNT_KEY\\"   --name $SOURCE_CONTAINER | jq -r \\".publicAccess\\"\`

# set original permission
az storage container set-permission   --name $DESTINATION_CONTAINER   --account-name \\"$DESTINATION_ACCOUNT_NAME\\"   --account-key \\"$DESTINATION_ACCOUNT_KEY\\"   --public-access $PUBLIC_ACCESS

",
            ],
            "env": Array [
              Object {
                "name": "SOURCE_ACCOUNT_NAME",
                "valueFrom": Object {
                  "secretKeyRef": Object {
                    "key": "azurestorageaccountname",
                    "name": "azure-somecoolprojectprod-volume",
                  },
                },
              },
              Object {
                "name": "SOURCE_ACCOUNT_KEY",
                "valueFrom": Object {
                  "secretKeyRef": Object {
                    "key": "azurestorageaccountkey",
                    "name": "azure-somecoolprojectprod-volume",
                  },
                },
              },
              Object {
                "name": "DESTINATION_ACCOUNT_NAME",
                "valueFrom": Object {
                  "secretKeyRef": Object {
                    "key": "azurestorageaccountname",
                    "name": "azure-somecoolprojectdev-volume",
                  },
                },
              },
              Object {
                "name": "DESTINATION_ACCOUNT_KEY",
                "valueFrom": Object {
                  "secretKeyRef": Object {
                    "key": "azurestorageaccountkey",
                    "name": "azure-somecoolprojectdev-volume",
                  },
                },
              },
              Object {
                "name": "SOURCE_CONTAINER",
                "value": "prod-files1",
              },
              Object {
                "name": "DESTINATION_CONTAINER",
                "value": "dev-files2",
              },
            ],
            "envFrom": Array [],
            "image": "mcr.microsoft.com/azure-cli:2.15.1",
            "imagePullPolicy": "IfNotPresent",
            "name": "restore-db",
            "resources": Object {
              "limits": Object {
                "cpu": "300m",
                "memory": "512Mi",
              },
              "requests": Object {
                "cpu": "50m",
                "memory": "64Mi",
              },
            },
          },
        ],
        "restartPolicy": "Never",
      },
    },
    "ttlSecondsAfterFinished": 86400,
  },
}
`;

exports[`should create restore job from prod to prod 1`] = `
Object {
  "apiVersion": "batch/v1",
  "kind": "Job",
  "metadata": Object {
    "annotations": Object {
      "kapp.k14s.io/update-strategy": "skip",
    },
    "labels": Object {
      "application": "www",
      "component": "restore-container",
      "owner": "my-team",
      "team": "my-team",
    },
    "name": "restore-container-my-test-branch-47n9d8",
    "namespace": "some-cool-project-secret",
  },
  "spec": Object {
    "backoffLimit": 0,
    "template": Object {
      "metadata": Object {
        "annotations": Object {
          "kapp.k14s.io/deploy-logs": "for-new-or-existing",
        },
      },
      "spec": Object {
        "containers": Array [
          Object {
            "command": Array [
              "sh",
              "-c",
              "

[ ! -z $SOURCE_CONTAINER ] || (echo \\"No SOURCE_CONTAINER\\"; exit 1)
[ ! -z $SOURCE_ACCOUNT_NAME ] || (echo \\"No SOURCE_ACCOUNT_NAME\\"; exit 1)
[ ! -z $SOURCE_ACCOUNT_KEY ] || (echo \\"No SOURCE_ACCOUNT_KEY\\"; exit 1)
[ ! -z $DESTINATION_CONTAINER ] || (echo \\"No DESTINATION_CONTAINER\\"; exit 1)
[ ! -z $DESTINATION_ACCOUNT_NAME ] || (echo \\"No DESTINATION_ACCOUNT_NAME\\"; exit 1)
[ ! -z $DESTINATION_ACCOUNT_KEY ] || (echo \\"No DESTINATION_ACCOUNT_KEY\\"; exit 1)


echo \\"starting restore container $SOURCE_CONTAINER into $DESTINATION_CONTAINER\\"

# copy container
az storage copy -r   --source-account-name \\"$SOURCE_ACCOUNT_NAME\\"   --source-account-key \\"$SOURCE_ACCOUNT_KEY\\"   --source-container \\"$SOURCE_CONTAINER\\"   --account-name \\"$DESTINATION_ACCOUNT_NAME\\"   --account-key \\"$DESTINATION_ACCOUNT_KEY\\"   --destination-container \\"$DESTINATION_CONTAINER\\"

# get original permission
PUBLIC_ACCESS=\`az storage container show-permission   --account-name \\"$SOURCE_ACCOUNT_NAME\\"   --account-key \\"$SOURCE_ACCOUNT_KEY\\"   --name $SOURCE_CONTAINER | jq -r \\".publicAccess\\"\`

# set original permission
az storage container set-permission   --name $DESTINATION_CONTAINER   --account-name \\"$DESTINATION_ACCOUNT_NAME\\"   --account-key \\"$DESTINATION_ACCOUNT_KEY\\"   --public-access $PUBLIC_ACCESS

",
            ],
            "env": Array [
              Object {
                "name": "SOURCE_ACCOUNT_NAME",
                "valueFrom": Object {
                  "secretKeyRef": Object {
                    "key": "azurestorageaccountname",
                    "name": "azure-somecoolprojectprod-volume",
                  },
                },
              },
              Object {
                "name": "SOURCE_ACCOUNT_KEY",
                "valueFrom": Object {
                  "secretKeyRef": Object {
                    "key": "azurestorageaccountkey",
                    "name": "azure-somecoolprojectprod-volume",
                  },
                },
              },
              Object {
                "name": "DESTINATION_ACCOUNT_NAME",
                "valueFrom": Object {
                  "secretKeyRef": Object {
                    "key": "azurestorageaccountname",
                    "name": "azure-somecoolprojectprod-volume",
                  },
                },
              },
              Object {
                "name": "DESTINATION_ACCOUNT_KEY",
                "valueFrom": Object {
                  "secretKeyRef": Object {
                    "key": "azurestorageaccountkey",
                    "name": "azure-somecoolprojectprod-volume",
                  },
                },
              },
              Object {
                "name": "SOURCE_CONTAINER",
                "value": "prod-files1",
              },
              Object {
                "name": "DESTINATION_CONTAINER",
                "value": "prod-files2",
              },
            ],
            "envFrom": Array [],
            "image": "mcr.microsoft.com/azure-cli:2.15.1",
            "imagePullPolicy": "IfNotPresent",
            "name": "restore-db",
            "resources": Object {
              "limits": Object {
                "cpu": "300m",
                "memory": "512Mi",
              },
              "requests": Object {
                "cpu": "50m",
                "memory": "64Mi",
              },
            },
          },
        ],
        "restartPolicy": "Never",
      },
    },
    "ttlSecondsAfterFinished": 86400,
  },
}
`;
